### PR TITLE
Archive tree improvements

### PR DESCRIPTION
**Pull request by gestchild**
The current ArchiveTree is making way too many api requests, this fixes that and puts in typing and some tests.
Once the API contains `totalParts` etc. we'll be able to remove more API calls

### CODE REVIEW

**Comment by jamesgorrie** [comment]
Do you think it's worth rebasing out some of those commits?

File: common/model/catalogue.js [hunk 1]
```diff
@@ -4,6 +4,18 @@ export type Work = {
   ...Object,
 };
 
+// type Work = {|
```
Comment (jamesgorrie) [inline common/model/catalogue.js:7]: Why'd we comment this out?

File: common/test/utils/works.test.js [hunk 2]
```diff
@@ -5,8 +5,9 @@ import {
   getItemsWith,
   getItemIdentifiersWith,
   getWorkIdentifiersWith,
+  getArchiveAncestorArray,
 } from '../../utils/works';
-import { WorkFixture } from '../fixtures/catalogueApi/work';
+import { WorkFixture, workWithPartOf } from '../fixtures/catalogueApi/work';
```
Comment (jamesgorrie) [inline common/test/utils/works.test.js:10]: We're using different capitilisations here, is there a reason?

File: common/utils/works.js [hunk 3]
```diff
@@ -16,16 +16,16 @@ type DownloadImage = {|
   height: ?number,
 |};
 
-export type ArchiveNode = {|
+export type NodeWork = {|
```
Comment (jamesgorrie) [inline common/utils/works.js:19]: What is a `NodeWork`? I wonder if we have something similar in the API?

File: common/utils/works.js [hunk 4]
```diff
@@ -246,28 +246,27 @@ export function getItemIdentifiersWith(
...
+    [...partOfArray, parsePartOf(nextPart)],
+    nextPart.partOf &&
+      nextPart.partOf.find(part => {
+        return nextPart.referenceNumber.includes(part.referenceNumber);
+      })
+  );
+}
+
+export function getArchiveAncestorArray(work: Work): NodeWork[] {
+  return makeArchiveAncestorArray([], work.partOf && work.partOf[0]).reverse();
```
Comment (davidpmccormick) [inline common/utils/works.js:271]: would it make more sense to have the `.reverse()` as part of the `makeArchiveAncestorArray` function?

File: common/views/components/ArchiveBreadcrumb/ArchiveBreadcrumb.js [hunk 5]
```diff
@@ -82,9 +82,10 @@ type Props = {|
 |};
 
 const ArchiveBreadcrumb = ({ work }: Props) => {
-  const ancestorArray = getAncestorArray(work);
-  const firstCrumb = ancestorArray[0];
-  const middleCrumbs = ancestorArray.length > 1 ? ancestorArray.slice(1) : [];
+  const archiveAncestorArray = getArchiveAncestorArray(work);
```
Comment (davidpmccormick) [inline common/views/components/ArchiveBreadcrumb/ArchiveBreadcrumb.js:85]: Oh, I see – it wouldn't necessarily need to be reversed. Ok ignore me.

File: common/model/catalogue.js (hunk 1 above)
Comment (gestchild) [reply common/model/catalogue.js:7]: Didn't comment it out, I put it in there as a comment. It's essentially a duplicate of the type on line 2, which is being used everywhere, but needs filling out with what's in here. Just for a different PR I think.

File: common/test/utils/works.test.js (hunk 2 above)
Comment (gestchild) [reply common/test/utils/works.test.js:10]: no reason, I'll make it consistent

**Review by davidpmccormick — APPROVED** [review]
I'm down with it

**Pull request merged**